presto-main/src/main/java/com/facebook/presto/operator/scalar/MapSubscriptOperator.java

```diff
@@ -202,6 +206,13 @@ public static Object subscript(boolean legacyMissingKey, MethodHandle keyEqualsM
         if (legacyMissingKey) {
             return null;
         }
-        throw new PrestoException(INVALID_FUNCTION_ARGUMENT, format("%s not present in map", key));
+        throw new PrestoException(INVALID_FUNCTION_ARGUMENT, format("%s not present in map", toObjectValue(keyType, key, session)));
+    }
+
+    private static Object toObjectValue(Type type, Object value, ConnectorSession session)
```

Comment:
Put the value at the end of the message: `"Key not present in map: " + value`

Or just say `"Key not present in map"`